Домашнее задание 5: Изменение данных (+темная тема)

softwarrior opened the pull request:
## 5. Изменение данных
  - [x] Можно выбрать записи кликом по checkbox-ам. По клику на чекбокс в шапке выбираются и сбрасываются записи только на текущей странице.
  - [x] Если применили какие то фильтры или отсортировали таблицу, выбранные записи сбрасываются
  - [x] По кнопке удалить, записи удаляются из redux, выбор сбрасывается.
  - [x] По кнопке изменить статус, меняются статусы в redux.
  - [x] По клику на ряд, открывается модалка с редактированием записи. Если ввести код 123 и нажать сохранить, запись обновляется в таблице
      - [x] Можно менять ФИО, Статус заказа
      - [x] Если поле ФИО пустой, появляется ошибка и поле подсвечвается красным.
      - [x] Если код введен неверно отображается ошибка и поле подсвечивается красным. (Общий компонент поддеживает состояние ошибки)
      - [x] Когда форма открыта все остальное становится некликабельным пока форма не будет закрыта

## 6. Бонусы
  - [x] Темная тема

@Muslik Заранее большое спасибо за ревью!

Commit:
Домашнее задание 4: Redux time

Commit:
Merge branch 'homework3' into homework4

Commit:
сделал debounce ввод поиска

Commit:
добавил поиск по номеру заказа

Commit:
Merge branch 'master' into homework4

* master:

Commit:
рефакторинг

Commit:
вынес стейты из страницы

Commit:
переделал пагинацию и создал общие для страницы константы

Commit:
ввод пагинации из инпута, замена моков

Commit:
исправил отображение в таблице

Commit:
поправил цвета у статуса

Commit:
поправил редакс замечания, сортировку, фильтрацию, пагинацию

Commit:
добавил открытие формы

Commit:
сделал выделение

Commit:
сделал удаление

Commit:
изменение статуса

Commit:
форма для редактирования заказов

Muslik reviewed (changes requested):
Будет видео. jsx -> js

Commit:
Переименовал файлы из jsx в js там где нет компонент

Commit:
Избавился от функции isIncludeNumberOrString

Commit:
Селектор getOrders стал возвращать объект { orders, totalOrdersCount }

Commit:
Правка футера таблицы
  - Исправил цвет текста "Выбрано записей"
  - Текст и кнопки появляются только после выбора заказа.
  - Добавил функции для склонения слова "запись", "записи", "записей"
  - После изменения статуса закрывается дропдаун и убирается выделение

Commit:
Сделал сброс checked в радио группе после ее закрытия

Commit:
Правка формы
  - OrderForm расставил отступы и скобочки
  - Убрал лишние useMemo
  - Исправил цвет иконки у disabled инпута
  - После изменения статуса закрывается дропдаун
  - Сообщение "Неправильный код" появляется после ввода
  - Убрал функцию errorChecker
  - Исправил логику отображения ошибки
  - Убрал кнопку у Input если стиль disabled
  - Добавил дропдаун и логику выхода из формы
  - Input одиночный тег
  - У кнопки добавил disabled

softwarrior commented:
## ДЗ-5 исправления:

#### Переименование файлов
  - [x] Переименовал файлы из jsx в js там где нет компонент

#### Правка селекторов
  - [x] Избавился от функции isIncludeNumberOrString
  - [x] Селектор getOrders стал возвращать объект { orders, totalOrdersCount }

#### Правка футера таблицы
  - [x] Исправил цвет текста "Выбрано записей"
  - [x] Текст и кнопки появляются только после выбора заказа.
  - [x] Добавил функции для склонения слова "запись", "записи", "записей"
  - [x] После изменения статуса закрывается дропдаун и убирается выделение
  - [x] Сделал сброс checked в радио группе после ее закрытия

#### Правка формы
  - [x] OrderForm расставил отступы и скобочки
  - [x] Убрал лишние useMemo
  - [x] Исправил цвет иконки у disabled инпута
  - [x] После изменения статуса закрывается дропдаун
  - [x] Сообщение "Неправильный код" появляется после ввода
  - [x] Убрал функцию errorChecker
  - [x] Исправил логику отображения ошибки
  - [x] Убрал кнопку у Input если стиль disabled
  - [x] Добавил дропдаун и логику выхода из формы
  - [x] Input одиночный тег
  - [x] У кнопки добавил disabled

#### Правка в Redux
  - [x] Reducer changeOrder теперь принимает массив параметров

##### Примечание:
ФИО по английски потому что https://json-generator.com/ умеет генерировать только английские имена.

**Исправил все замечания из видео.**

@Muslik  **Большое спасибо за отличное ревью!**
